core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4976,14 +4993,49 @@ private static void computeVersion(ServletContext context) {
      * Version number of this Jenkins.
      */
     public static String VERSION = UNCOMPUTED_VERSION;
-
+    
+    /**
+     * Minimal Java level required to run this Jenkins.
+     * If {@code null}, the version is unknown.
+     */
+    private static @CheckForNull String JAVA_MIN_LEVEL = null;
+    private static VersionNumber DEFAULT_JAVA_MIN_LEVEL = new VersionNumber("8");
```

Comment:
> they can mark the requirement as a system property

There must be a better approach than that for this, ideally something integrating with plugin metadata, e.g. defined properties in the POM, or manifest entries, so we can publish PRE requirements via update center -- I think there was a change to maven-hpi-plugin proposed a while back.

"Install (or update) a plugin and see Jenkins not starting up again" seems like not a great experience.